jax/_src/util.py

```diff
@@ -378,6 +378,10 @@ def wrap_name(name, transform_name):
 def fun_name(fun: Callable):
   return getattr(fun, "__name__", "<unnamed function>")
 
+def fun_qual_name(fun: Callable):
```

Comment:
Should we just always use `__qualname__`?

Reply:
You mean: never use the `fun_name`? Not clear. There are at least a couple of tests that verify that we are using `__name__` implicitly, and I wanted the least disruptive change.